src/main/java/frc/robot/util/SparkAdapter.java

```diff
@@ -153,15 +171,22 @@ public SparkAdapter(
    * factory methods. It assumes the SparkFlex object is already configured or will be configured
    * approriately by the public constructors or factory methods.
    *
+   * @param namePrefix The prefix for the log entries.
    * @param spark The SparkFlex object to adapt.
    */
-  private SparkAdapter(SparkFlex spark) {
+  private SparkAdapter(String namePrefix, SparkFlex spark) {
+    this.namePrefix = namePrefix;
     this.spark = new SparkFlexAccessor(spark);
+
+    String name = String.format("%s/SparkMax-%d", namePrefix, spark.getDeviceId());
+    this.logOutputCurrent = new DoubleLogEntry(LOG, "%s/OutputCurrent", name);
+    this.logTemperature = new DoubleLogEntry(LOG, "%s/Temperature", name);
```

Comment:
Did you really intend for `name` to be set as metadata on these logs? (that's what the constructor's 3rd param is)
```suggestion
    this.logOutputCurrent = new DoubleLogEntry(LOG, name + "/OutputCurrent");
    this.logTemperature = new DoubleLogEntry(LOG, name + "/Temperature");
```